rust-runtime/aws-smithy-runtime-api/src/client/interceptors/context/wrappers.rs

```diff
@@ -158,6 +158,15 @@ declare_wrapper!(
     (response: Response)
 );
 
+impl<'a, I, O, E: Debug> BeforeDeserializationInterceptorContextMut<'a, I, O, E> {
+    /// Downgrade this helper struct, returning the underlying InterceptorContext. There's no good
+    /// reason to use this unless you're writing tests or you have to interact with an API that
+    /// doesn't support the helper structs.
+    pub fn into_inner(&mut self) -> &'_ mut InterceptorContext<I, O, E> {
+        self.inner
+    }
+}
```

Comment:
Perhaps we should modify the RetryClassifier trait to just take what it needs from the context rather than the whole context.

Reply:
Maybe, although I like passing the whole thing because it feels more future-proof. However, will we ever need to check anything other than the response pre- and post-serialization? I can't think of why we would.

I'm OK making this change but I'd rather do it as a separate PR.

Reply:
Maybe make this `#[doc(hidden)]` in the mean time.